src/elements/rating/rating.js

```diff
@@ -0,0 +1,111 @@
+(function() {
+  'use strict';
+
+  angular
+    .module('semantic.ui.elements.rating', [])
+    .constant('ratingConfig', {
+      max: 5,
+      stateActive: null,
+      stateHover: null,
+      stateHoverParent: null
+    })
+    .directive('smRating', smRating);
+
+  var template =
+    '<div ng-mouseleave="reset()" ng-keydown="onKeydown($event)" tabindex="0" role="slider"' +
+      'aria-valuemin="0" aria-valuemax="{{range.length}}" aria-valuenow="{{value}}"' +
+      'class="ui rating" ng-class="hoverValue > -1 && (stateHoverParent || \'selected\')">' +
+      '<i ng-repeat="r in range track by $index" ng-mouseenter="enter($index + 1)" ng-click="rate($index + 1)"' +
+        'class="icon" ng-class="[' +
+          '($index < value && (r.stateActive || \'active\')) || \'\',' +
+          '($index < hoverValue && (r.stateHover || \'selected\')) || \'\']">' +
+      '</i>' +
+    '</div>';
+
+  function smRating() {
+    return {
+      restrict:'EA',
+      replace: true,
+      require: ['smRating', 'ngModel'],
+      scope: {
```

Comment:
on second thought, the new scope is okay because of `controllerAs`, and since this isn't a transcluding directive (should it be?), the isolate scope is fine